components/notification-sender-config/org.wso2.carbon.identity.notification.sender.tenant.config/src/main/java/org/wso2/carbon/identity/notification/sender/tenant/config/NotificationSenderManagementServiceImpl.java

```diff
@@ -232,25 +265,60 @@ public List<EmailSenderDTO> getEmailSenders() throws NotificationSenderManagemen
         }
     }
 
+    @Override
+    public List<SMSSenderDTO> getSMSSenders(boolean inheritTenantSettings)
+            throws NotificationSenderManagementException {
+
+        return retrieveSMSSenders(inheritTenantSettings);
+    }
+
     @Override
     public List<SMSSenderDTO> getSMSSenders() throws NotificationSenderManagementException {
 
+        return retrieveSMSSenders(true);
+    }
+
+    private List<SMSSenderDTO> retrieveSMSSenders(boolean inheritTenantSettings)
+            throws NotificationSenderManagementException {
+
         try {
-            Resources publisherResources = NotificationSenderTenantConfigDataHolder.getInstance()
-                    .getConfigurationManager()
-                    .getResourcesByType(PUBLISHER_RESOURCE_TYPE);
-            List<Resource> smsPublisherResources = publisherResources.getResources().stream().filter(resource ->
-                    resource.getAttributes().stream().anyMatch(attribute ->
-                            PUBLISHER_TYPE_PROPERTY.equals(attribute.getKey()) &&
-                                    SMS_PUBLISHER_TYPE.equals(attribute.getValue()))).collect(Collectors.toList());
-            return smsPublisherResources.stream().map(NotificationSenderUtils::buildSmsSenderFromResource).collect(
-                    Collectors.toList());
+            String tenantDomain = PrivilegedCarbonContext.getThreadLocalCarbonContext().getTenantDomain();
+            Resources publisherResources = getPublisherResources(tenantDomain, inheritTenantSettings);
```

Comment:
As the tenant domain can be fetched from the thread local, you may don't need to pass the tenant domain as a method argument.

Reply:
Fixed